mavros/scripts/mavsys

```diff
@@ -88,9 +92,8 @@ def main():
 
     mode_args = subarg.add_parser('mode', help="Set mode")
     mode_args.set_defaults(func=do_mode)
-    mode_group = mode_args.add_mutually_exclusive_group(required=True)
-    mode_group.add_argument('-b', '--base-mode', type=int, help="Base mode code")
-    mode_group.add_argument('-c', '--custom-mode', type=str, help="Custom mode string (same as in ~/state topic)")
+    mode_args.add_argument('-b', '--base-mode', type=int, help="Base mode code")
+    mode_args.add_argument('-c', '--custom-mode', type=str, help="Custom mode string (same as in ~/state topic)")
```

Comment:
Document where you need to specify base mode with custom mode.

Reply:
https://github.com/mavlink/mavros/pull/810#issuecomment-329759400

Reply:
Base mode 3 is weird. CUSTOM_MODE_ENABLED + TEST_ENABLED.
More likely to see CUSTOM_MODE + SAFETY_ARMING + AUTO_ENABLED.
But last time i checked APM it do not rely on base_mode at all (except custom mode flag).

If PX4 now require AUTO_ENABLED just add quirk to handler. Nothing more.

Reply:
And `GCS_Common.cpp` still operates same way:
```cpp
1289 /*
1290   handle a SET_MODE MAVLink message
1291  */
1292 void GCS_MAVLINK::handle_set_mode(mavlink_message_t* msg)
1293 {
1294     uint8_t result = MAV_RESULT_FAILED;
1295     mavlink_set_mode_t packet;
1296     mavlink_msg_set_mode_decode(msg, &packet);
1297 
1298     // only accept custom modes because there is no easy mapping from Mavlink flight modes to AC flight modes
1299     if (packet.base_mode & MAV_MODE_FLAG_CUSTOM_MODE_ENABLED) {
1300         if (set_mode(packet.custom_mode)) {
1301             result = MAV_RESULT_ACCEPTED;
1302         }
1303     } else if (packet.base_mode == MAV_MODE_FLAG_DECODE_POSITION_SAFETY) {
1304         // set the safety switch position. Must be in a command by itself
1305         if (packet.custom_mode == 0) {
1306             // turn safety off (pwm outputs flow to the motors)
1307             hal.rcout->force_safety_off();
1308             result = MAV_RESULT_ACCEPTED;
1309         } else if (packet.custom_mode == 1) {
1310             // turn safety on (no pwm outputs to the motors)
1311             if (hal.rcout->force_safety_on()) {
1312                 result = MAV_RESULT_ACCEPTED;
1313             }
1314         }
1315     }
1316 
1317     // send ACK or NAK
1318     mavlink_msg_command_ack_send_buf(msg, chan, MAVLINK_MSG_ID_SET_MODE, result);
1319 }
```

Reply:
Ok so if one wants to set different flags, where should it be set? In https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/sys_status.cpp#L931-L945 ?

Reply:
https://www.youtube.com/watch?v=8zVYru974iQ

(Something went wrong with gtk-recordmydesktop, video goes too fast...)